### PR TITLE
sometimes propagate JobTimeoutSec=0 into JobRunningTimeoutSec=

### PR DESCRIPTION
**Pull request by poettering**
A fix for #6402. Not pretty, but should work.

### CODE REVIEW

File: src/core/load-fragment.c [hunk 1]
```diff
@@ -4165,6 +4165,80 @@ int config_parse_protect_system(
...
+        assert(rvalue);
+        assert(u);
+
+        r = parse_sec_fix_0(rvalue, &usec);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse JobTimeoutSec= parameter, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        if (usec == USEC_INFINITY) {
```
Comment (keszybz) [inline src/core/load-fragment.c]: This should solve the immediate issue, but I had the same thought as @Werkov in https://github.com/systemd/systemd/issues/6402#issuecomment-333564159: shouldn't this `if` be dropped`? This patch restores compatibility for the specific case of timeout=infinity, but not for other values.

**Comment by poettering** [comment]
Force pushed a new version making the suggested change. PTAL

File: src/core/load-fragment.c [hunk 2]
```diff
@@ -4145,6 +4145,78 @@ int config_parse_protect_system(
...
+        assert(u);
+
+        r = parse_sec_fix_0(rvalue, &usec);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse JobTimeoutSec= parameter, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        /* If the user explicitly changed JobTimeoutSec= also change JobRunningTimeoutSec=, for compatibility with old
+         * versions. If JobRunningTimeoutSec= was explicitly set, avoid this however as whatever the usec picked should
```
Comment (joukewitteveen) [inline src/core/load-fragment.c:4176]: s/usec/user/

Comment (keszybz) [reply src/core/load-fragment.c:4176]: Fixed in merge.

**Pull request merged**